ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.js

```diff
@@ -0,0 +1,349 @@
+  const metaMaskAccounts = useSelector(getMetaMaskAccounts);
+  const connectRequests = useSelector(
+    (state) => state.metamask.institutionalFeatures?.connectRequests,
+  );
+  const [isLoading, setIsLoading] = useState(false);
+  const [tokenAccounts, setTokenAccounts] = useState([]);
+  const [error, setError] = useState(false);
+  const t = useI18nContext();
+  const [copied, handleCopy] = useCopyToClipboard();
+  const {
+    removeAddTokenConnectRequest,
+    setCustodianNewRefreshToken,
+    getCustodianAccounts,
+  } = mmiActions;
+  const connectRequest = connectRequests ? connectRequests[0] : undefined;
+
+  useEffect(() => {
+    isMountedRef.current = true;
+    return () => (isMountedRef.current = false);
+  }, []);
+
+  useEffect(() => {
+    const getTokenAccounts = async () => {
+      if (!connectRequest) {
+        history.push(mostRecentOverviewPage);
+        return;
+      }
+
+      try {
+        const custodianAccounts = await dispatch(
+          getCustodianAccounts(
+            connectRequest.token,
+            connectRequest.apiUrl,
+            connectRequest.service,
+            false,
+          ),
+        );
+
+        const filteredAccounts = custodianAccounts.filter(
+          (account) => metaMaskAccounts[account.address.toLowerCase()],
+        );
+
+        const mappedAccounts = filteredAccounts.map((account) => ({
+          address: account.address,
+          name: account.name,
+          labels: account.labels,
+          balance:
+            metaMaskAccounts[account.address.toLowerCase()]?.balance || 0,
+        }));
+
+        if (isMountedRef.current) {
+          setTokenAccounts(mappedAccounts);
+          setIsLoading(false);
+        }
+      } catch (e) {
+        console.log(e);
+        setError(true);
+        setIsLoading(false);
+      }
+    };
+
+    getTokenAccounts();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  if (!connectRequest) {
+    history.push(mostRecentOverviewPage);
+    return null;
+  }
+
+  const onRemoveAddTokenConnectRequest = ({ origin, apiUrl, token }) => {
+    dispatch(
+      removeAddTokenConnectRequest({
+        origin,
+        apiUrl,
+        token,
+      }),
+    );
+  };
+
+  const handleReject = () => {
+    onRemoveAddTokenConnectRequest(connectRequest);
+    history.push(mostRecentOverviewPage);
+  };
+
+  const handleApprove = async () => {
+    if (error) {
+      global.platform.openTab({
+        url,
+      });
+      handleReject();
+      return;
+    }
+
+    setIsLoading(true);
+
+    try {
+      await Promise.all(
+        tokenAccounts.map(async (account) => {
+          await dispatch(
+            setCustodianNewRefreshToken({
+              address: account.address,
+              newAuthDetails: {
+                refreshToken: connectRequest.token,
+                refreshTokenUrl: connectRequest.apiUrl,
+              },
+            }),
```

Comment:
Why do we need to await this, just curiouss?

Reply:
We need this done before we route to the next page/show the banner?

Reply:
Yes, we need it because `setCustodianNewRefreshToken` is async. And we want to wait until all accounts have changed their refresh token before navigating to the new page. 

If you are curious, here's a link with more info:
https://whimsical.com/token-refresh-TcL2qyu2YpM9K8LRjq9ga4